nixos/modules/security/acme/doc.xml

```diff
@@ -249,18 +249,19 @@ systemd.services.dns-rfc2136-conf = {
   path = [ pkgs.bind ];
   script = ''
     mkdir -p /var/lib/secrets
+    chmod 755 /var/lib/secrets
     tsig-keygen rfc2136key.example.com &gt; /var/lib/secrets/dnskeys.conf
     chown named:root /var/lib/secrets/dnskeys.conf
     chmod 400 /var/lib/secrets/dnskeys.conf
 
-    # Copy the secret value from the dnskeys.conf, and put it in
-    # RFC2136_TSIG_SECRET below
+    # extract secret value from the dnskeys.conf
+    while read x y; do if [ "$x" = "secret" ]; then secret="''${y:1:''${#y}-3}"; fi; done &lt; /var/lib/secrets/dnskeys.conf
 
     cat &gt; /var/lib/secrets/certs.secret &lt;&lt; EOF
     RFC2136_NAMESERVER='127.0.0.1:53'
     RFC2136_TSIG_ALGORITHM='hmac-sha256.'
     RFC2136_TSIG_KEY='rfc2136key.example.com'
-    RFC2136_TSIG_SECRET='your secret key'
+    RFC2136_TSIG_SECRET='$secret'
```

Comment:
Slight change: Should be using double quotes here since you want the var from the shell

Reply:
Yes, this would not work at all due to this -- single quotes don't expand anything.

Reply:
It works with single quotes. That's how it is in the version I'm using. 

The single quotes don't interfere with $secret expanding.  bash expands variables in that whole line because it's within a  <<EOF section

Reply:
Ah I totally missed that!

Reply:
I thought you had found a bug until I looked at my in-production working version, and saw that it was working with single quotes. Then I figured out why it works this way :)